Unite.Web/Client/src/domain/_shared/components/upload/UploadDialog.vue

```diff
@@ -0,0 +1,171 @@
+<template>
+  <q-dialog
+    v-model="dialog"
+    @keyup.esc="dialog = false"
+    @hide="onClose"
+    persistent>
+
+    <q-card style="min-width: 300px">
+      <q-card-section>
+        <div class="text-h6">Upload {{ subjectLower }} data</div>
+      </q-card-section>
+
+      <q-card-section class="q-gutter-y-sm">
+        <div class="row q-gutter-x-sm">
+          <div class="text-body1">File type:</div>
+          <q-radio v-model="fileType" val="json" label="JSON" dense />
+          <q-radio v-model="fileType" val="tsv" label="TSV" dense />
+        </div>
+
+        <div class="row">
+          <div v-if="fileType === 'json'" class="col">
+            Download json file <a :href="templatePathJson" :download="`${subjectLower}-template.json`" class="u-link">template</a>
+          </div>
+          <div v-if="fileType === 'tsv'" class="col">
+            Download tsv file <a :href="templatePathTsv" :download="`${subjectLower}-template.tsv`" class="u-link">template</a>
+          </div>
+        </div>
+
+        <q-file
+          label="File"
+          v-model="file.value"
+          :rules="file.rules"
+          @rejected="notifyError(errorMessage)"
+          clearable
+          square outlined dense>
+          <template #prepend>
+            <q-icon name="las la-paperclip" size="sm" />
+          </template>
+        </q-file>
+      </q-card-section>
+
+      <q-card-actions align="right" class="text-primary">
+        <q-btn
+          label="Cancel"
+          dense flat no-caps
+          v-close-popup
+        />
+
+        <q-btn
+          label="Apply"
+          :disable="!canApply"
+          @click="onApply"
+          dense flat no-caps
+          v-close-popup
+        />
+      </q-card-actions>
+    </q-card>
+  </q-dialog>
+</template>
+
+<script>
+const defaultErrorMessage = 'Something went terribly wrong, sorry!';
```

Comment:
I like the message, but we should probably write something less horrifying for users :)

Reply:
Removed this error as it was obsolete